pandas/core/base.py

```diff
@@ -92,14 +116,17 @@ def __unicode__(self):
         return object.__repr__(self)
 
     def _dir_additions(self):
+        # type: () -> typing.Set[str]
```

Comment:
In what cases would we use `str` over `Text` (see `__unicode__`) in type annotations?

Reading the docs [1] I think what you are doing here is probably correct because `str` is python 2/3 implementation specific but I'm not 100% sure.


[1]
https://docs.python.org/3/library/typing.html#typing.Text